### PR TITLE
adding list type

### PR DESCRIPTION
**Pull request by sinemetu1**
I updated the tests, although I don't think any of the classes under /emr-dynamodb-hadoop/src/test/java/org/apache/hadoop/dynamodb/ are covered.

### CODE REVIEW

**Comment by sinemetu1** [comment]
I created a release for preferring list over set here: https://github.com/sinemetu1/emr-dynamodb-connector/releases/tag/prefer-list I'm not sure of a clean way to map hive array types to these two dynamo types. There needs to be some kind of hint I guess.

**Comment by sinemetu1** [comment]
I cleaned up the pr. I don't really know what to do with these classes [here](https://github.com/awslabs/emr-dynamodb-connector/pull/71/files#diff-2f30800692c81e292a24bf83fff783d7R22) and [here](https://github.com/awslabs/emr-dynamodb-connector/pull/71/files#diff-ab0d3a13cbdbe9355c8ffea90ac3d94bR22), but I chose to create a `DynamoDBListStorageHandler`. I'm happy to make changes to get this merged in.

**Comment by gdoron** [comment]
@sinemetu1 Thanks **a lot** for this!
Could have saved me a lot of time in the past if I had it!

Any chance you'll be able to ad  Map as well?

File: emr-dynamodb-hadoop/src/test/java/org/apache/hadoop/dynamodb/test/DynamoDBTestUtils.java [hunk 1]
```diff
@@ -70,6 +70,21 @@ public static String getRandomString() {
     return strings;
   }
 
+  public static List<AttributeValue> getRandomStringsAndNumbers() {
```
Comment (itsgrimetime) [inline emr-dynamodb-hadoop/src/test/java/org/apache/hadoop/dynamodb/test/DynamoDBTestUtils.java]: Is this method used anywhere?

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBSerDe.java [hunk 2]
```diff
@@ -90,6 +92,11 @@ public ObjectInspector getObjectInspector() throws SerDeException {
     return Text.class;
   }
 
+
+  protected HiveDynamoDBType getTypeObjectFromHiveType(String type) {
```
Comment (itsgrimetime) [inline emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBSerDe.java]: What's the point in extracting this as a method?

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBStorageHandler.java [hunk 3]
```diff
@@ -237,14 +237,22 @@ public void configureOutputJobProperties(TableDesc tableDesc, Map<String, String
     configureTableJobProperties(tableDesc, jobProperties);
   }
 
+  protected boolean isHiveDynamoDBItemMapType(String type){
```
Comment (itsgrimetime) [inline emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBStorageHandler.java]: Same here, what is the purpose in wrapping these in another method call?

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/type/HiveDynamoDBTypeUtil.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ */
+package org.apache.hadoop.hive.dynamodb.type;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import org.apache.hadoop.hive.serde2.lazy.LazyDouble;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class HiveDynamoDBTypeUtil {
```
Comment (itsgrimetime) [inline emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/type/HiveDynamoDBTypeUtil.java]: Could you add some tests for this class?

Comment (sinemetu1) [reply emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/type/HiveDynamoDBTypeUtil.java]: Added tests [here](https://github.com/awslabs/emr-dynamodb-connector/pull/71/files#diff-457b7d995e711ccd87d71de9b6d72d94R17).

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBSerDe.java (hunk 2 above)
Comment (sinemetu1) [reply emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBSerDe.java]: It's so we can use the [ListTypeFactor here](https://github.com/awslabs/emr-dynamodb-connector/pull/71/files#diff-01feb7d63da4958c25fa3fb6ba334509R9).

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBStorageHandler.java (hunk 3 above)
Comment (sinemetu1) [reply emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBStorageHandler.java]: Similar situation where we need to [prefer list type factory](https://github.com/awslabs/emr-dynamodb-connector/pull/71/files#diff-f3a6ef9c1c04a0cfe168d14061163fe3R21)

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBSerDe.java (hunk 2 above)
Comment (itsgrimetime) [reply emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBSerDe.java]: ah makes sense

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBStorageHandler.java (hunk 3 above)
Comment (itsgrimetime) [reply emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBStorageHandler.java]: 👍

File: emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/type/HiveDynamoDBTypeUtil.java (hunk 4 above)
Comment (itsgrimetime) [reply emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/type/HiveDynamoDBTypeUtil.java]: whoops, sorry, I overlooked those somehow. thanks for pointing them out.

**Pull request merged**